--- a/libmariadb/libmariadb/CMakeLists.txt
+++ b/libmariadb/libmariadb/CMakeLists.txt
@@ -407,6 +407,7 @@ IF(CMAKE_SIZEOF_VOID_P EQUAL 8 AND MSVC)
 ENDIF()
 
 IF(CMAKE_SYSTEM_NAME MATCHES "Linux" OR
+   CMAKE_SYSTEM_NAME MATCHES "kFreeBSD" OR
    CMAKE_SYSTEM_NAME MATCHES "GNU")
   IF (NOT WITH_ASAN AND NOT WITH_TSAN AND NOT WITH_UBSAN AND NOT WITH_MSAN)
     TARGET_LINK_LIBRARIES (libmariadb "-Wl,--no-undefined")
